src/ui/auth/uaa/uaa.go

```diff
@@ -43,51 +38,85 @@ func GetClient() (uaa.Client, error) {
 		Endpoint:      UAASettings.Endpoint,
 		SkipTLSVerify: !UAASettings.VerifyCert,
 	}
-	client, err = uaa.NewDefaultClient(cfg)
-	return client, err
+	return uaa.NewDefaultClient(cfg)
+}
+
+// Auth is the implementation of AuthenticateHelper to access uaa for authentication.
+type Auth struct {
+	sync.Mutex
+	client uaa.Client
 }
 
-func doAuth(username, password string, client uaa.Client) (*models.User, error) {
-	t, err := client.PasswordAuth(username, password)
+//Authenticate ...
+func (u *Auth) Authenticate(m models.AuthModel) (*models.User, error) {
+	if err := u.ensureClient(); err != nil {
+		return nil, err
+	}
+	t, err := u.client.PasswordAuth(m.Principal, m.Password)
 	if t != nil && err == nil {
 		//TODO: See if it's possible to get more information from token.
-		u := &models.User{
-			Username: username,
+		user := &models.User{
+			Username: m.Principal,
 			Password: "1234567ab",
-			Email:    username + "@placeholder.com",
-			Realname: username,
-		}
-		err = dao.OnBoardUser(u)
-		if err == nil {
-			return u, nil
+			Email:    m.Principal + "@uaa.placeholder",
```

Comment:
1) Is it possible to take email address  "test@example.com" as  UAA's principal if it is true, do not create email address
2) Is it possible that username or email contains any leading or trailing space?

Reply:
I'll open a bug to better handle the Email.